superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -319,6 +320,18 @@ function dbReducer(
         [action.payload.name]: action.payload.json,
       };
     case ActionType.textChange:
+      if (action.payload.name === 'query_input') {
```

Comment:
in this case, because it has a different return value, should we just make a different action name?

Reply:
Ah, good point. Yeah, makes sense.